testing/integration/otel_test.go

```diff
@@ -874,123 +770,169 @@ func TestOtelFBReceiverE2E(t *testing.T) {
-			if err != nil {
-				t.Logf("No otel configuration file at %s", otelConfigPath)
-				return
-			}
-			t.Logf("Contents of otel config file:\n%s\n", string(contents))
-		}
-	})
-	// Now we can actually create the fixture and run it
-	fixture, err := define.NewFixtureFromLocalBuild(t, define.Version(), aTesting.WithAdditionalArgs([]string{"--config", otelConfigPath}))
-	require.NoError(t, err)
 
 	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(5*time.Minute))
 	defer cancel()
-	err = fixture.Prepare(ctx, fakeComponent)
+	err = fixture.Prepare(ctx)
 	require.NoError(t, err)
 
-	var fixtureWg sync.WaitGroup
-	fixtureWg.Add(1)
-	go func() {
-		defer fixtureWg.Done()
-		err = fixture.RunOtelWithClient(ctx)
-	}()
+	err = fixture.Configure(ctx, configBuffer.Bytes())
 
-	// Make sure find the logs
-	actualHits := &struct{ Hits int }{}
+	cmd, err := fixture.PrepareAgentCommand(ctx, nil)
+	require.NoError(t, err, "cannot prepare Elastic-Agent command: %w", err)
+
+	output := strings.Builder{}
+	cmd.Stderr = &output
+	cmd.Stdout = &output
+
+	err = cmd.Start()
+	require.NoError(t, err)
+
+	// Make sure the Elastic-Agent process is not running before
+	// exiting the test
+	t.Cleanup(func() {
+		// Ignore the error because we cancelled the context,
+		// and that always returns an error
+		_ = cmd.Wait()
+		if t.Failed() {
+			t.Log("Elastic-Agent output:")
+			t.Log(output.String())
+		}
+	})
+
+	var agentDocs estools.Documents
+	var otelDocs estools.Documents
+
+	// Make sure to find the logs
+	// and assert document equivalency for docs ingested by both classic and otel mode
 	require.Eventually(t,
 		func() bool {
 			findCtx, findCancel := context.WithTimeout(context.Background(), 10*time.Second)
 			defer findCancel()
 
-			docs, err := estools.GetLogsForIndexWithContext(findCtx, info.ESClient, ".ds-"+index+"*", map[string]interface{}{
+			agentDocs, err = estools.GetLogsForIndexWithContext(findCtx, info.ESClient, ".ds-logs-e2e-"+info.Namespace+"*", map[string]interface{}{
 				"log.file.path": inputFilePath,
 			})
-			require.NoError(t, err)
 
-			actualHits.Hits = docs.Hits.Total.Value
-			return actualHits.Hits == numEvents
+			// Note: OTel docs are sent to namespace = "otel" + info.Namespace
+			otelDocs, err = estools.GetLogsForIndexWithContext(findCtx, info.ESClient, ".ds-logs-e2e-otel"+info.Namespace+"*", map[string]interface{}{
+				"log.file.path": inputFilePath,
+			})
+
+			return (agentDocs.Hits.Total.Value == numEvents) && (otelDocs.Hits.Total.Value == numEvents)
 		},
 		2*time.Minute, 1*time.Second,
-		"Expected %d logs, got %v", numEvents, actualHits)
+		"Expected documents to be indexed by classic agent and otel mode")
+
+	doc1 := agentDocs.Hits.Hits[0].Source
+	doc2 := otelDocs.Hits.Hits[0].Source
+	ignoredFields := []string{
+		// Expected to change between filebeat and fbreceiver
+		"@timestamp",
+		"agent.ephemeral_id",
+		"agent.id",
+		"agent.version",
```

Comment:
I assume `agent.version` should match in both documents now that we have the otel runtime. Can you confirm that?